### PR TITLE
WIP - Bps 744 create case functional tests

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-744

### Change description ###
Introduced CreateTestCase

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    void setup() throws Exception {
+
+        dmUrl = dmUploadService.uploadToDmStore(
+            "Certificate.pdf",
+            "documents/supplementary-evidence.pdf"
+        );
+    }
+
+    @Test
+    public void should_create_case_upon_exception_record() throws Exception {
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: may be `should_create_case_from_valid_exception_record` ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            SampleData.JURSIDICTION,
+            SampleData.JURSIDICTION + "_" + CreateExceptionRecord.CASE_TYPE,
+            ImmutableMap.of(
+                "case.poBox", randomPoBox.toString()
+            )
+        );
+        return caseDetailsList.stream().findFirst();
+    }
+
+    private Boolean isCaseCreated(String caseCcdId, CaseDetails exceptionRecord, int expectedScannedDocsSize) {
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: are we expecting more test cases with different number of documents ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        List<ScannedDocument> updatedScannedDocuments = getScannedDocuments(updatedCase);
+
+        return updatedScannedDocuments.size() == expectedScannedDocsSize;
+    }
+
+    private void verifyCaseIsCreated(
+        String caseCcdId,
+        CaseDetails exceptionRecord
+    ) {
+        CaseDetails createdCase = ccdApi.getCase(
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: You are retrieving the case twice ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 2 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: This has been fixed

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 1 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Done

**Pull request closed without merging**